hist/hist/src/TH1.cxx

```diff
@@ -3800,16 +3800,16 @@ TFitResultPtr TH1::Fit(const char *fname ,Option_t *option ,Option_t *goption, D
    Int_t ndim=GetDimension();
    if (linear){
       if (ndim<2){
-         TF1 f1(fname, fname, xxmin, xxmax);
-         return Fit(&f1,option,goption,xxmin,xxmax);
+         f1=new TF1(fname, fname, xxmin, xxmax);
+         return Fit(f1,option,goption,xxmin,xxmax);
       }
       else if (ndim<3){
-         TF2 f2(fname, fname);
-         return Fit(&f2,option,goption,xxmin,xxmax);
+         f2=new TF2(fname, fname);
+         return Fit(f2,option,goption,xxmin,xxmax);
       }
       else{
-         TF3 f3(fname, fname);
-         return Fit(&f3,option,goption,xxmin,xxmax);
+         f3=new TF3(fname, fname);
+         return Fit(f3,option,goption,xxmin,xxmax);
       }
```

Comment:
@amadio Is it correct changes?
There was commit https://github.com/root-project/root/commit/610a87c from @lmoneta which was changing exact this place. Now you reverting changes back. Is it intended?

Reply:
As the commit message says, I just reverted it to test this PR without showing the errors/warnings caused by this commit.